### ISSUE TITLE
Chunk Load Thrashing

### ISSUE THREAD

**Issue by GreenMarine** [NillerMedDild/Enigmatica2Expert#1194]
**Modpack version**
1.64a

**Can the issue be reproduced?**
Yes

**Logs**
https://pastebin.com/mnS2zeyc

**Issue**
Hello! I'm back. After talking to the EnderIO guys here:

https://github.com/SleepyTrousers/EnderIO/issues/5131

Which was a result of this discussion here:

https://github.com/NillerMedDild/Enigmatica2Expert/issues/1190

We determined that a likely cause of the EnderIO network lag is chunk load thrashing. Apparently when chunks are loaded and unloaded, the network rebuilds itself. I ran tellme with a monitor on chunk loading and got a fair amount of rapid chunk load / unloads.

Some of these can be seen here: https://pastebin.com/mnS2zeyc
and a lot more here: https://pastebin.com/2ByXAtgv

Logs were taken with:
tellme track enable all-dims chunk-load
tellme track enable all-dims chunk-unload

This unloading / loading cycles at a high rate and happens even if users are offline.

Loaded dims:

[tellme]: DIM    0: overworld        [ 358 loaded chunks,  118 loaded entities, 0 players]
[tellme]: DIM -343800852: Spectre          [   0 loaded chunks,    0 loaded entities, 0 players]
[tellme]: DIM    2: Storage Cell     [   0 loaded chunks,    0 loaded entities, 0 players]

(The loaded chunks count in dim 0 will cycle fairly quickly.)

Examining entities in the thrashing chunks indicates nothing particularly suspect.

**Comment by GreenMarine** [comment]
Closing this. I couldn't figure out the problem, but migrated the server to an i9. Solve it with more compute!

**Comment by NillerMedDild** [comment]
Reopening in case someone can help here, as everyone stand to benefit from better performance :D

**Comment by Ricket** [comment]
The rapid chunk load/unloads might be the issue I opened here: https://github.com/MinecraftForge/MinecraftForge/issues/5406

I was able to help relieve the issue by increasing the forgeChunkLoading.cfg `dormantChunkCacheSize`. Effectively making it so that these chunks are only thrashing into and out of the dormant cache, not into/out of disk.

The comments on that issue _might_ indicate that this issue is fixed in MC 1.13.1, I'm not certain.

**Comment by Ricket** [comment]
BTW you mentioned dimensions at the end -- if you are also having dimension loading thrash then there's a similar variable in forge.cfg for dimensions: `dimensionUnloadQueueDelay`. You could try increasing that to make a dimension stay loaded longer, which would reduce thrash.

And of course I am not recommending specific numbers for these configs because it's very dependent on the situation and the amount of server memory (and CPU to some degree).

**Comment by GreenMarine** [comment]
The dimension trashing was due to one of the player chunkloading part of the nether - which might have then triggered this chunk load/unload problem. What's described in that issue you linked does match what we see on our server. In fact, several times when I teleported to a problematic chunk I found a cow or other entity on the edge of it. It struck me at the time, but I didn't have a model for linking that to the problem. Now I have that model.
